shell.nix

```
@@ -0,0 +1,13 @@
+{ nixpkgs ? import <nixpkgs> {} }:
+let
+  inherit (nixpkgs) pkgs;
+  inherit (pkgs) haskellPackages;
```

`nixpkgs.pkgs == nixpkgs`, it just recursively re-exposes itself.

```suggestion
  inherit (nixpkgs) haskellPackages;
```

---

So `inherit` just gives us access to a bunch of attributes belonging to some JSON object `nixpkgs` here? And the added line you suggested will put those attributes into `haskellPackages` so you access them via `haskellpackages.x` where `x` is an attribute?

Oh and also `nixpkgs` are like a set of specific "stores" which specify some specific set of builds for various software/dependencies? Kind of like which snapshot you want in Stack? And instead of using `nixpkgs` I can use `pkgsStatic` if I'm not using `TemplateHaskell` and that'll make it so these builds are suitable for distribution (being statically-linked) which is pretty much my goal with `Nix`, here?